0000-notes-on-oer-encoding/0000-notes-on-oer-encoding.md

```diff
@@ -0,0 +1,285 @@
+---
+title: Notes on OER Encoding
+draft: 1
+---
+
+# Introduction
+
+The Interledger stack consists of a collection of different protocols. In order to keep some consistency between these separate, but related protocols, the Interledger team has decided to use some common choices for encoding formats across at least those protocols we control.
+
+These choices will not be applicable for all protocols in the Interledger ecosystem. For example, we used a binary encoding while a text-based encoding may be more suitable. We used an encoding with a fixed schema, while a schema-less encoding may be easier to extend by third-parties.
+
+Nevertheless, designers of new Interledger protocols SHOULD consider staying within the boundaries set by this document. Deviations from these norms should be carefully considered. This is intended to minimize the load on implementors who should be able to write components such as date encoding and decoding once and reuse it across protocols.
+
+# How to use this document
+
+This document is intended as a first reference for implementors to use when writing code that serializes and deserializes Interledger formats. We intend to make this document sufficient on its own, such that reading the official ASN.1/OER specifications is not required to correctly implement Interledger formats.
+
+Implementors SHOULD NOT aim to create a full-featured OER parser, since Interledger intentionally does not use some of the more complex features of ASN.1. Implementors MAY use a full-featured CANONICAL-OER implementation, but we expect most will prefer a lighter-weight implementation that only handles types that are actually used in the Interledger protocols.
+
+# Foundation
+
+Interledger encodings are based on and compliant with Abstract Syntax Notation One (ASN.1) using Canonical Octet Encoding Rules (CANONICAL-OER).
+
+## ASN.1
+
+### What is ASN.1?
+
+Abstract Syntax Notation One (ASN.1) is an international standard for describing abstract data types. ITU gives the following definition:
+
+> Abstract Syntax Notation number One
+> is a standard that defines a formalism
+> for the specification of abstract data types.
+
+In plain English, ASN.1 describes what your data can be, without making any statements about how it could be encoded in bits and bytes.
+
+### Why ASN.1?
```

Comment:
A lot of developers have bad associations with ASN.1 and I remember you ascribing most of that to BER and DER. It might be worth mentioning your argument that OER basically fixes most of the things people hated before about the ASN.1 family